node_modules/@npmcli/package-json/lib/read-package.js

```diff
@@ -0,0 +1,39 @@
+// This is JUST the code needed to open a package.json file and parse it.
+// It's isolated out so that code needing to parse a package.json file can do so in the same way as this module does, without needing to require the whole module, or needing to require the underlying parsing library.
+
+const { readFile } = require('fs/promises')
+const parseJSON = require('json-parse-even-better-errors')
+
+async function read (filename) {
+  try {
+    const data = await readFile(filename, 'utf8')
+    return data
+  } catch (err) {
+    err.message = `Could not read package.json: ${err}`
+    throw err
+  }
+}
+
+function parse (data) {
+  try {
+    const content = parseJSON(data)
+    return content
+  } catch (err) {
+    err.message = `Invalid package.json: ${err}`
```

Comment:
[Question for an earlier PR I guess]

Do we want to include the stack, etc in the error message, or just prefix the error message?

```suggestion
    err.message = `Invalid package.json: ${err.message}`
```

Reply:
This question belongs in https://github.com/npm/package-json